### PR TITLE
Branch cleanup

### PR DESCRIPTION
**Pull request by Quazia**
Should cover the majority of changes in M2.

Once all the true-ups and changes are in branch-cleanup and all CI is passing we can bring it into dev.

### CODE REVIEW

File: apps/range-voting/app/script.js [hunk 1]
```diff
@@ -38,8 +38,16 @@ async function handleEvents(response) {
...
-    allocations = app.external(response.returnValues.addr, AllocationJSON.abi)
+    let funcSig = response.returnValues.funcSig
+    console.info('[RangeVoting > script]: received ExternalContract', funcSig)
+    // Should actually be a case-switch
+    if(funcSig == 'f1d12a23'){
+      console.log('Loading Projects Data')
+      resolve(loadVoteDataProjects(voteData, voteId))
+    } else {
+      console.log('Loading Allocations Contract')
+      allocations = app.external(response.returnValues.addr, AllocationJSON.abi)
```
Comment (sohkai) [inline apps/range-voting/app/script.js]: Is the range voting app only meant to be used with one allocations contract at a time?

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  contentBorderActive,
+  contentBorder,
+  textSecondary,
+} = theme
+
+class OptionsInputDropdown extends React.Component {
+  static propTypes = {
+    input: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    value: PropTypes.array.isRequired,
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: We are using this intuitive generic handler to react to input state changes on every input we use, dynamically reading the name and value of each monitored input:
`  changeField = ({ target: { name, value } }) => {
    this.setState({ [name]: value })
  }`

In this case, if we rename the prop variable to `values` it improves its semantics but we then are bound to add another extra handler just for this Input, I carefully handled every input on the app to behave that way.

What is your opinion in this case? Is it better to stick to semantics or just comment the functionality and let it that way?

File: apps/allocations/contracts/Allocations.sol [hunk 3]
```diff
@@ -89,22 +97,29 @@ contract Allocations is AragonApp, Fundable { // solium-disable-line blank-lines
     event FundAccount(uint256 accountId);
     event SetDistribution(uint256 payoutId, uint256 amount);
 
+    function () public payable { 
+        revert();
+    }
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/contracts/Allocations.sol]: Solved at 8eff31a1688e3f96a4b1e880f539c52d585aa805

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+
+  addOption = () => {
+    // TODO: Implement some rules about what an 'Option can be' duplicates, etc
+    const { input, name, value } = this.props
+    if (input && !value.includes(input) && this.props.validator(input)) {
+      this.props.onChange({ target: { name, value: [...value, input] } })
+      this.props.onChange({ target: { name: 'optionsInput', value: '' } })
+      this.props.error = true
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Fixed at 680cb16a4d0113719a95b845dab7af8b25ecd542

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  textSecondary,
+} = theme
+
+class OptionsInputDropdown extends React.Component {
+  static propTypes = {
+    input: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    value: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+    validator: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at 680cb16a4d0113719a95b845dab7af8b25ecd542

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    entities: PropTypes.object,
+  }
+
+
+  addOption = () => {
+    // TODO: Implement some rules about what an 'Option can be' duplicates, etc
+    const { input, name, value } = this.props
+    if (input && !value.includes(input) && this.props.validator(input)) {
+      this.props.onChange({ target: { name, value: [...value, input] } })
+      this.props.onChange({ target: { name: 'optionsInput', value: '' } })
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at 21ccd527f30b45171a1afc5b132910b607777826

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  removeOption = option => {
+    const { name, value } = this.props
+    let index = value.indexOf(option)
+    // Double exclamation to make sure is reemoved
+    !!value.splice(index, 1) &&
+      this.props.onChange({
+        target: { name, value },
+      })
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at 21ccd527f30b45171a1afc5b132910b607777826

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+class OptionsInputDropdown extends React.Component {
+  static propTypes = {
+    input: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    value: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+    validator: PropTypes.func,
+    error: PropTypes.bool,
+    entities: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at dbc0752666b669ce72ee2d9d944d92ed06a35aca

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  onChangeInput = (index, items) => {
+    let value = this.props.entities[index].addr
+    this.props.onChange({ target: { name: 'optionsInput', value } })
+  }
+
+  render() {
+    const loadOptions = this.props.value.map(option => (
+      <div className="option" key={option}>
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at cb907961775934422e7dcadf332321d1fb9566f1

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved at 2cf1b066b6dcae9ca9ecfcda249177a5f2f4128e

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+const StyledOptionsInput = styled.div`
+  display: flex;
+  flex-direction: column;
+  > .option {
+    display: flex;
+    margin-bottom: 0.625rem;
+    > :first-child {
+      flex-grow: 1;
+    }
+    > svg {
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Solved both at 24120848640fa6869313b2616588a4c1e11dbd94 👍

File: apps/allocations/app/components/Form/Input/OptionsInputDropdown.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+      </div>
+    ))
+    return (
+      <StyledOptionsInput empty={!this.props.input.length}>
+        {loadOptions}
+        <div className="option">
+          <DropDown
+            items={this.props.entities.map(entity => entity.data.name)}
+            onChange={this.onChangeInput}
+            wide={true}
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/OptionsInputDropdown.js]: Fixed at OptionsInputDropdown by using activeItem prop: 540a42f3fb5b0c8aeab5bfae568620110abce23a and at DropDown active prop: e81daa622b5b508a690cff98ae274218c49f862e

File: apps/allocations/app/components/Form/Input/SettingsInput.js [hunk 12]
```diff
@@ -6,11 +6,28 @@ import { Badge, Text, theme } from '@aragon/ui'
...
+    onChange: PropTypes.func.isRequired,
+  }
+  
+  changeChecked = (checked) => {
+    const {name} = this.props
+    const value = !this.state.checked
+    this.setState({checked: value})
+    this.props.onChange({ target: { name: name, value: value } })
+  }
+  state = { checked: false }
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/SettingsInput.js]: Fixed at: b9f1c4aad0ee71f211c9f8c4dfe48e26f946f725 (lately converted to SFC so now it does not have state at all)

File: apps/allocations/app/components/Form/Input/SettingsInput.js [hunk 13]
```diff
@@ -6,11 +6,28 @@ import { Badge, Text, theme } from '@aragon/ui'
...
+    name: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+  }
+  
+  changeChecked = (checked) => {
+    const {name} = this.props
+    const value = !this.state.checked
+    this.setState({checked: value})
+    this.props.onChange({ target: { name: name, value: value } })
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Form/Input/SettingsInput.js]: Solved at: 39fdbb5342e99ed3cc8324d73b8b2caead312aed

File: apps/allocations/app/components/Panel/NewAllocation.js [hunk 14]
```diff
@@ -183,37 +214,47 @@ class NewAllocation extends React.Component {
...
+                />
+              }
+            />
+          }
+          {this.state.addressSetting == false &&
+            <FormField
+              separator
+              label={this.state.addressSetting}
+              input={
+                <OptionsInput
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/app/components/Panel/NewAllocation.js]: Fixed at ce33a8a4dc21bb87e601617b4f350bf35fc3511a by adding a `visible` prop to FormField so we have granular control over the conditions

File: apps/projects/app/components/Form/Input/SettingsInput.js [hunk 15]
```diff
@@ -6,11 +6,17 @@ import { Badge, Text, theme } from '@aragon/ui'
 import { CheckBox } from '../../Shared'
 
 class SettingsInput extends React.Component {
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/projects/app/components/Form/Input/SettingsInput.js]: Solved at b275d179c9d5396a9ce3e9d5de53527c70cb1fce

File: apps/projects/app/components/Shared/assets/components/NewProject.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+
+const NewProject = props => (
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/projects/app/components/Shared/assets/components/NewProject.js]: Solved at 7d2b359f941a067368f5d5fcdbbb88361895a5a0

File: apps/projects/app/components/Shared/assets/components/ArrowDown.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+const ArrowDown = props => (
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/projects/app/components/Shared/assets/components/ArrowDown.js]: Done at 05f034a5fb66d12dcc667b6f9c2f0d08b70141f0

File: apps/projects/app/index.html [hunk 18]
```diff
@@ -1,14 +1,12 @@
 <!DOCTYPE html>
 <html>
-
-<head>
+  <head>
+    <script src="http://localhost:8097"></script>
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/projects/app/index.html]: done -> 985d4362eb16b0cd6117cce41a480a245eda5097

File: apps/allocations/contracts/Allocations.sol [hunk 19]
```diff
@@ -139,7 +167,43 @@ contract Allocations is AragonApp, Fundable { // solium-disable-line blank-lines
...
+
+    function fund(uint256 id) external payable { 
+        Payout storage payout = payouts[id];
+        require(!payout.informational);
+        payout.balance = payout.balance.add(msg.value);
+        emit FundAccount(id);
+    }
+
+    function runPayout(uint256 _payoutId) external payable isInitialized returns(bool success) {
```
[the remark this answers is not in the document]
Comment (ottodevs) [reply apps/allocations/contracts/Allocations.sol]: done

**Pull request closed without merging**